### PR TITLE
show programmatic config of spring otlp exporter

### PR DESCRIPTION
**Pull request by zeitlinger**

### CODE REVIEW

File: spring-native/src/main/java/io/opentelemetry/example/graal/OtlpExporterConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+  @Bean
+  ConfigurableSpanExporterProvider otlpSpanExporterProvider() {
+    return new OtlpSpanExporterProvider();
+  }
+
+  private static class OtlpSpanExporterProvider
+      implements ConfigurableSpanExporterProvider, AutoConfigureListener {
+
+    private final AtomicReference<MeterProvider> meterProviderRef =
```
Comment (jeanbisutti) [inline spring-native/src/main/java/io/opentelemetry/example/graal/OtlpExporterConfiguration.java:36]: Could you explain the concurrency issue?

File: spring-native/src/main/java/io/opentelemetry/example/graal/Application.java [hunk 2]
```diff
@@ -24,4 +26,10 @@ public AutoConfigurationCustomizerProvider otelCustomizer() {
                     .drop(SemanticAttributes.URL_PATH, "^/actuator")
                     .build());
   }
+
+  @Bean
+  OpenTelemetryInjector registerGlobalOpenTelemetry() {
+    // not recommended, since you can just auto-wire OpenTelemetry where you need it
```
[the remark this answers is not in the document]
Comment (zeitlinger) [reply spring-native/src/main/java/io/opentelemetry/example/graal/Application.java:32]: As said, it's not recommended and usually not necessary. 
However, if users want to user the global, then we should tell them how to do it.

Would you rather have it in a different example?

File: spring-native/src/main/java/io/opentelemetry/example/graal/OtlpExporterConfiguration.java (hunk 1 above)
Comment (zeitlinger) [reply spring-native/src/main/java/io/opentelemetry/example/graal/OtlpExporterConfiguration.java:36]: No concurrency issue - this is just copied from the SDK auto config.

File: spring-native/src/main/java/io/opentelemetry/example/graal/Application.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (jeanbisutti) [reply spring-native/src/main/java/io/opentelemetry/example/graal/Application.java:32]: Global has caused several bugs  in the logging and JDBC instrumentations and with the OpenTelemetry starter. These issues have been fixed by removing global and injecting an `OpenTelemetry` instance into the instrumentations.

In addition, dependency injection is something extensively used in the Spring ecosystem.

So, I would not suggest to use global in an example.

**Comment by zeitlinger** [comment]
@jeanbisutti can you check again?

**Comment by jeanbisutti** [comment]
LGTM Thanks!

**Pull request merged**